### PR TITLE
CBL-5030: Opening db of v2 into db of VersionVector causes conversion…

### PR DESCRIPTION
**Pull request by jianminzhao**
… of all documents.

For the legacy database of huge size, we cannot afford to enumerate and convert all documents as we open it into the database of current versioning. Fortunately, we are able to handle documents of all earlier versions, specifically TreeVersioning_v2 and TreeVersioning, in the database of current VectorVersioning. The program will upgrade those documents to the current versioning at appropriate time on demand.

### CODE REVIEW

File: LiteCore/Database/DatabaseImpl+Upgrade.cc [hunk 1]
```diff
@@ -110,6 +110,7 @@ namespace litecore {
             }
         }
         LogTo(DBLog, "\t%" PRIu64 " documents upgraded, now committing changes...", docCount);
+        DebugAssert(docCount == 0, "Should not upgrade all documents when openning the database.");
```
Comment (snej) [inline LiteCore/Database/DatabaseImpl+Upgrade.cc:113]: Why is this here? The assertion will fail unless the database is empty or already uses version vectors.

Since you are commenting out the only call to this method, why not just delete it? Actually you can delete the whole source file, I think.

File: C/c4Database.cc [hunk 2]
```diff
@@ -73,7 +73,7 @@ static void ensureConfigDirExists(const C4DatabaseConfig2& config) {
 
 static C4DatabaseConfig newToOldConfig(const C4DatabaseConfig2& config2) {
     return C4DatabaseConfig{config2.flags | kC4DB_AutoCompact, NULL,
-                            (config2.flags & kC4DB_VersionVectors) ? kC4VectorVersioning : kC4TreeVersioning_v2,
+                            (config2.flags & kC4DB_VersionVectors) ? kC4VectorVersioning : kC4TreeVersioning,
```
Comment (snej) [inline C/c4Database.cc:76]: This is really strange -- I did not think we were using kC4TreeVersioning_v2 at all anymore. Looks like I did this by accident back in commit ea82fb755 in 2021.

File: LiteCore/Database/DatabaseImpl.cc [hunk 3]
```diff
@@ -193,8 +193,9 @@ namespace litecore {
 
         // Yup, mismatch confirmed, so deal with it:
         if ( versDoc.exists() ) {
-            // Existing db versioning does not match runtime config!
-            upgradeDocumentVersioning(curVersioning, newVersioning, transaction());
+            // Pass for existing database. We can handle documents of old versions in
```
Comment (snej) [inline LiteCore/Database/DatabaseImpl.cc:196]: Why keep this? Just delete the lines.

**Review by snej — CHANGES_REQUESTED** [review]

File: LiteCore/Database/DatabaseImpl+Upgrade.cc (hunk 1 above)
Comment (jianminzhao) [reply LiteCore/Database/DatabaseImpl+Upgrade.cc:113]: Yes, I can simply delete that function. Perhaps I first #if it out and think to move it to cblite as a tool to upgrade db.

File: LiteCore/Database/DatabaseImpl.cc (hunk 3 above)
Comment (jianminzhao) [reply LiteCore/Database/DatabaseImpl.cc:196]: okay, will do it.

**Comment by jianminzhao** [comment]
We. have tests to open dbs of all legacy versions, tree_v2, tree.

File: LiteCore/Database/DatabaseImpl+Upgrade.cc (hunk 1 above)
Comment (jianminzhao) [reply LiteCore/Database/DatabaseImpl+Upgrade.cc:113]: Perhaps I can just delete it and file a ticket to move it to cblite that mentions the commit where the function is deleted.

**Review by snej — APPROVED** [review]
